extensions/wikidata/tests/src/org/openrefine/wikidata/editing/EditBatchProcessorTest.java

```diff
@@ -161,11 +168,70 @@ public void testMultipleBatches()
         }
     }
 
+    @Test
+    public void testMultipleBatchesMediaInfo()
+            throws MediaWikiApiErrorException, InterruptedException, IOException {
+        // Prepare test data
+        MonolingualTextValue label = Datamodel.makeMonolingualTextValue("village in Nepal", "en");
+        List<MonolingualTextValue> labels = Collections.singletonList(label);
+        StatementGroup statement = Datamodel.makeStatementGroup(Collections.emptyList());
+        List<StatementGroup> statements = Collections.singletonList(statement);
+        TermUpdate labelsUpdate = Datamodel.makeTermUpdate(null, Collections.emptyList());
+        List<String> ids = new ArrayList<>();
+        for (int i = 124; i < 190; i++) {
+            ids.add("Q" + String.valueOf(i));
+        }
+        List<MediaInfoIdValue> mids = ids.stream().map(e -> Datamodel.makeWikimediaCommonsMediaInfoIdValue(e))
+                .collect(Collectors.toList());
+        List<TermedStatementEntityUpdate> batch = mids.stream()
+                .map(mid -> new TermedStatementEntityUpdateBuilder(mid).addLabel(label, false).build())
+                .collect(Collectors.toList());
+
+        int batchSize = 50;
+        List<MediaInfoDocument> fullBatch = mids.stream()
+                .map(mid -> Datamodel.makeMediaInfoDocument(mid, labels, statements).findLabel(label.toString())
+                        .build()).collect(Collectors.toList());
```

Comment:
I am not sure why you are using the `findLabel` method here. Why not just `Datamodel.makeMediaInfoDocument(mid, labels, statements)` (without the `.build()` call, since this is not a builder class?)